### DISCUSSION TITLE
Config file exists in project root directory but seems ignored

### DISCUSSION THREAD

**Discussion by mloskot** [terraform-linters/tflint#1706] [Q&A] [answered]
### Summary

The `tflint --help` says `Config file name (default: .tflint.hcl)` which I take as meaning that `.tflint.hcl` is looked up for and read automatically by default.

There is `.tflint.hcl` exists in top-level folder of my Terraform project, but `tflint` seems to be ignoring it unless I specify it explicitly.

### Command

`tflint` but see below for details.

### Terraform Configuration

```terraform
N/A
```

### TFLint Configuration

```HCL
plugin "terraform" {
  enabled = true
  preset  = "recommended"
}

plugin "azurerm" {
    enabled = true
    version = "0.21.0"
    source  = "github.com/terraform-linters/tflint-ruleset-azurerm"
}

rule "terraform_typed_variables" {
  enabled = false
}
```

### Output

Run with implicit/automatic `.tflint.hcl` loading - expected output `terraform_typed_variables` warnings filtered out

```console
$ tflint --format compact --recursive | grep terraform_typed_variables | wc -l
87
```

Run with explicit `.tflint.hcl` via filename only - expected output `terraform_typed_variables` warnings filtered out

```console
$ tflint --format compact --recursive --config .tflint.hcl | grep terraform_typed_variables | wc -l
87
```

Run with explicit `.tflint.hcl` via absolute path - Success! All `terraform_typed_variables` warnings filtered

```console
$ tflint --format compact --recursive --config $PWD/.tflint.hcl | grep terraform_typed_variables | wc -l
0
```

Interestingly, if I give the config file but via relative path, then I get this error:

```console
$ tflint --format compact --recursive --config ./.tflint.hcl
Failed to load TFLint config; failed to load file: open ./.tflint.hcl: no such file or directory
```

### TFLint Version

0.45.0

### Terraform Version

1.3.7

### Operating System

- [X] Linux
- [ ] macOS
- [X] Windows

**Comment by bendrucker** [comment, marked as answer]
https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/config.md#configuring-tflint

> However, if `--chdir` or `--recursive` is used, The config file in the changed directory will be loaded. 

I've updated this to clarify that this still applies to a config explicitly provided via a flag:

https://github.com/terraform-linters/tflint/commit/2d73106cf3e19c58451aeb948d2467440ca8f761

You can argue this behavior is confusing/awkward, but given the reverse it would be impossible to customize the name of the config file stored in each module.

**Reply by mloskot** [reply]
I did not figure the answer from the previous version of that document indeed. I appreciate the update, it makes it clear now. Thank you.
I see no need to argue the behaviour.
